### PR TITLE
[Spark-5889] Remove pid file after stopping service.

### PR DESCRIPTION
**Pull request by zhzhan**
Currently the pid file is not deleted, and potentially may cause some problem after service is stopped. The fix remove the pid file after service stopped.

### CODE REVIEW

File: sbin/spark-daemon.sh [hunk 1]
```diff
@@ -169,6 +169,7 @@ case $option in
       if [[ $(ps -p "$TARGET_ID" -o args=) =~ $command ]]; then
         echo "stopping $command"
         kill "$TARGET_ID"
```
Comment (srowen) [inline sbin/spark-daemon.sh]: Concretely, can there be an `if` statement here to check whether `kill` succeeded, and only `rm` if so?

Comment (andrewor14) [inline sbin/spark-daemon.sh]: a more concise way to do this is `kill ... && rm -rf $pid` I think

Comment (srowen) [inline sbin/spark-daemon.sh]: Yeah, it occurs to me that this doesn't even guarantee the process was killed, but that the signal was sent. Well, I kind of like `kill ... && rm -f $pid` and would vote for merging that.

**Comment by zhzhan** [comment]
@srowen @andrewor14 Changed to kill "$TARGET_ID" && rm -f $pid. Thanks.

**Comment by srowen** [comment]
@zhzhan I apologize for the extra round trip on this simple change, but it occurs to me that we should quote `"$pid"`. We should probably do it elsewhere in this an other scripts as @nchammas reminds us, but I imagine it's especially important as the argument to `rm`. Just in case there's... I don't know, a `/opt/very/important/Project` file and for some reason the PID path is `/opt/very/important/Project Data/pid`!

**Comment by srowen** [comment]
(Test failure looks like a Jenkins network problem; ignore it)

**Comment by zhzhan** [comment]
@srowen No problem. Thanks for the explanation, and that's a valid one.

**Pull request closed without merging**